marsha/core/tests/test_models.py

```diff
@@ -509,3 +513,61 @@ def test_playlist_access_uniqueness(self):
             playlist=PlaylistFactory(author=user, organization=OrganizationFactory()),
             user=user,
         )
+
+    def test_django_default_uniqueness(self):
+        """Verify that by default Django ignores our unique-together."""
+        user = UserFactory()
+        playlist = PlaylistFactory(author=user, organization=OrganizationFactory())
+
+        # fields that should be considered unique together, and force non-deleted objects
+        fields = {"user": user, "playlist": playlist, "deleted": None}
+
+        # create an object with these fields in database
+        playlist_access1 = PlaylistAccessFactory(**fields)
+
+        # create an object without saving it, with same "unique" fields
+        playlist_access2 = PlaylistAccess(**fields)
+
+        # here is the trick: we return nothing on our own method to let django do
+        # the default unique validation only
+        playlist_access2._get_conditional_non_deleted_unique_checks = (
+            lambda exclude=None: []
+        )
+
+        # so this should not raise because for django there is not unique_together on these fields
+        playlist_access2.validate_unique()
+
+        # same if one is already deleted which is the normal behavior
+        playlist_access1.delete(force_policy=SOFT_DELETE_CASCADE)
+        playlist_access2.validate_unique()
+
+        # and if new one is to be deleted
+        playlist_access2.deleted = now()
+        playlist_access2.validate_unique()
+
+    def test_django_enhanced_uniqueness(self):
+        """Verify that our unique-together defined by ConditionalUniqueIndex are checked."""
```

Comment:
s/ConditionalUniqueIndex/NonDeletedUniqueIndex right?

Reply:
Yes indeed (the reason is that I added `NonDeletedUniqueIndex` after ;) )